src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -522,6 +524,28 @@ ExtractResultRelationRTE(Query *query)
 }
 
 
+/*
+ * ForceExtractResultRelationRTE returns the table's resultRelation range table
+ * entry and errors out if there's no result relation at all, e.g. like in a
+ * SELECT query.
+ *
+ * This is a separate function (instead of using missingOk), so static analysis
+ * reasons about NULL returns correctly.
+ */
+RangeTblEntry *
+ForceExtractResultRelationRTE(Query *query)
```

Comment:
Could we choose an alternative name? When I read the current method, I can't guess the behaviour. `ExtractResultRelationRTEOrError`?